eip_operator/src/controller/pod.rs

```diff
@@ -25,66 +23,39 @@ impl k8s_controller::Context for Context {
     type Resource = Pod;
     type Error = Error;
 
-    const FINALIZER_NAME: &'static str = "eip.materialize.cloud/disassociate";
+    const FINALIZER_NAME: &'static str = "eip.materialize.cloud/disassociate_pod";
```

Comment:
I'm not sure changing this effects existing pods. I could leave it as is if there's an impact. This was primarily done for clarity and to allow the eip controller to also have a finalizer if needed in the future.

Reply:
Thanks, I might roll this back for now to simplify the pr. 

Yeah, sorry that was confusing. I guess I had thought about leaving the generic `eip.materialize.cloud/disassociate` open for the eip controller cleanup to handle, but that doesn't really make sense.